reference/shaders-msl-no-opt/vert/pass-array-by-value.vert

```diff
@@ -31,7 +31,7 @@ void spvArrayCopyFromConstant1(thread T (&dst)[N], constant T (&src)[N])
     for (uint i = 0; i < N; dst[i] = src[i], i++);
 }
 
-float4 consume_constant_arrays2(thread const float4 (&positions)[4], thread const float4 (&positions2)[4], thread int& Index1, thread int& Index2)
```

Comment:
The address space qualifiers are gone here. Is this intentional?

Reply:
Dug around to see why this is happening. This seems to be related to the `ensure_correct_attribute_type` changes made a while back.

https://github.com/KhronosGroup/SPIRV-Cross/blob/master/spirv_msl.cpp#L970

This dropping of references only happens on vertex shaders not fragment. I guess it's not a real problem since Input storage class is constant after all, but the pointer type is dropped from var.basetype, there might be some other problems related to that. I think we can fix that later.

Reply:
I've changed it to pass the pointer type to `ensure_correct_attribute_type()`. That will stop us from dropping that.